Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt

```diff
@@ -228,8 +233,15 @@ class DefaultZoomWindowController
             ToolType.IMPORTPNG.name,
             ToolType.SHAPE.name,
             ToolType.TEXT.name -> Constants.NOT_COMPATIBLE
-            ToolType.ERASER.name -> Constants.COMPATIBLE_ALL
-            else -> Constants.COMPATIBLE_NEW
+            ToolType.LINE.name,
+            ToolType.CURSOR.name,
+            ToolType.WATERCOLOR.name,
+            ToolType.SPRAY.name,
+            ToolType.BRUSH.name,
+            ToolType.ERASER.name,
+            ToolType.PIPETTE.name,
+            ToolType.SMUDGE.name -> Constants.COMPATIBLE_NEW
+            else -> Constants.COMPATIBLE_ALL
```

Comment:
There is every tool listed except the CLIP tool, which also does not function correctly now. It should work after adding it to the COMPATIBLE_NEW case. Since this would cover all tools now, the COMPATIBLE_ALL case can be removed completely. Please also rename COMPATIBLE_NEW ->  COMPATIBLE.

Reply:
Done. I also realized that it doesnt make sense to decide how things are drawn in the doDraw() depending on the zoom window compatibility. Now they don't depend on each other anymore and the code is a lot cleaner. Also all the tools work correctly now in terms of layer hierarchy and they obey the layer hierarchy now.

Reply:
Awesome!! Great idea! 💯  This totally makes sense. Functionality looks good as well now. 
One more little thing: the `doDraw` method is really long now, it would make sense to break it down in some private functions for readability.

Reply:
I cleaned up the doDraw() now for readability